### PR TITLE
[3.x] Split up quantity select into 2 variants

### PR DESCRIPTION
**Pull request by BobWez98**
V3 PR for #595

### CODE REVIEW

**Review by indykoning — APPROVED** [review]

File: resources/js/components/Product/QuantitySelect.vue [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        },
+
+        decrease() {
+            if (this.model - this.increment >= this.minQty) {
+                this.updateQty(this.model - this.increment)
+            }
+        },
+
+        updateQty(qty) {
+            let input = this.$scopedSlots.default()[0].context.$refs[this.inputRef]
```
Comment (royduin) [inline resources/js/components/Product/QuantitySelect.vue:42]: This should be possible without the ref, this is an anti pattern

File: resources/js/components/Product/QuantitySelect.vue [hunk 2]
```diff
@@ -0,0 +1,54 @@
+<script>
+export default {
+    props: {
+        inputRef: {
+            type: String,
+            default: 'qty-select-1',
+        },
+        defaultQty: {
```
Comment (royduin) [inline resources/js/components/Product/QuantitySelect.vue:8]: This isn't used?

File: resources/views/cart/item/quantity.blade.php [hunk 3]
```diff
@@ -4,23 +4,9 @@
...
-
-            ::min="Math.max(item.product.stock_item?.min_sale_qty, item.product.stock_item?.qty_increments) || null"
-            ::max="item.product.stock_item?.max_sale_qty"
-            ::step="item.product.stock_item?.qty_increments"
-        />
-        <x-rapidez::button type="submit" v-bind:dusk="'item-update-'+index" :title="__('Update')">
-            <x-heroicon-s-arrow-path class="h-4 w-4" />
-        </x-rapidez::button>
+    <div class="flex items-center gap-1">
+        <x-rapidez::quantity @change="mutate" index="item.id" model="variables.quantity" minSaleQty="(item.product.stock_item?.min_sale_qty ?? 1)" qtyIncrements="(item.product.stock_item?.qty_increments ?? 1)"/>
```
Comment (royduin) [inline resources/views/cart/item/quantity.blade.php:10]: The `max_sale_qty` is missing here

File: resources/views/components/quantity/enhanced.blade.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
+@props([
+    'model' => 'addToCart.qty',
+    'minSaleQty' => 1,
+    'qtyIncrements' => 1,
+    'index' => 1,
+])
+
+<quantity-select :input-ref="'qty-select-' + {{ $index }}" :model="parseInt({{ $model }})" :min-qty="{{ $minSaleQty }}" :increment="{{ $qtyIncrements }}" v-slot="qtySelect">
```
Comment (royduin) [inline resources/views/components/quantity/enhanced.blade.php:8]: This feels odd passing through the props

File: resources/views/components/quantity/enhanced.blade.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    <div class="flex">
+        <x-rapidez::button.outline @click.prevent="qtySelect.decrease">
```
Comment (royduin) [inline resources/views/components/quantity/enhanced.blade.php:10]: It isn't getting disabled when you can't decrease due the min qty

File: resources/views/components/quantity/enhanced.blade.php [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            <div class="bg-neutral h-0.5 w-3 hover:bg-white"></div>
+        </x-rapidez::button.outline>
+        <x-rapidez::input
+            name="qty"
+            type="number"
+            :label="false"
```
Comment (royduin) [inline resources/views/components/quantity/enhanced.blade.php:16]: This input isn't described now which is required for accessibility

File: resources/views/components/quantity/index.blade.php
Comment (royduin) [inline resources/views/components/quantity/index.blade.php:1]: This version isn't used, let's stick with 1 version.

File: resources/views/product/partials/grouped.blade.php [hunk 7]
```diff
@@ -17,7 +17,7 @@
                         @lang('Sorry! This product is currently out of stock.')
                     </p>
                 @else
-                    @include('rapidez::product.partials.quantity', ['product' => $groupedProduct])
+                    @include('rapidez::product.partials.quantity.default', ['product' => $groupedProduct])
```
Comment (royduin) [inline resources/views/product/partials/grouped.blade.php:20]: This file doesn't exist

**Review by royduin — CHANGES_REQUESTED** [review]
Thanks but I'm not really happy how this turns out, I tried to address some issues with https://github.com/rapidez/core/pull/645. Thoughts?

**Comment by royduin** [comment]
Closing this in favor of https://github.com/rapidez/core/pull/645

**Pull request closed without merging**